### PR TITLE
Preserve transform when using *Gradient:withOpacity

### PR DESCRIPTION
**Pull request by hyhugh**
i don't think it was intentional to lose it?

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA]. - i'm google employee but i'm not on flutter team.
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by goderbauer** [comment]
It does look like an oversight. Thanks for fixing it. Looks like it is failing some checks, though. Can you please take a look? Thank you.

**Comment by hyhugh** [comment]
I think the failed tests are unrelated to this CL, lemme wait a bit for head being green again.

**Comment by hyhugh** [comment]
@goderbauer just fixed some typos i made which made the tests failing, checks are passing now :) PTAL. thanks!

File: packages/flutter/test/painting/gradient_test.dart
Comment (nate-thegrate) [inline packages/flutter/test/painting/gradient_test.dart:473]:
```suggestion
    ));
```

**Review by nate-thegrate — COMMENTED** [review]
It feels a bit weird to fix `LinearGradient` whilst ignoring the same bug in `RadialGradient` & `SweepGradient`.

Do you think you could expand this fix to include the other `Gradient` subtypes?

File: packages/flutter/test/painting/gradient_test.dart
Comment (nate-thegrate) [reply packages/flutter/test/painting/gradient_test.dart:473]: (bonus points if you get rid of the weird comma in that test above this one too!)

Comment (hyhugh) [reply packages/flutter/test/painting/gradient_test.dart]: done :)

File: packages/flutter/test/painting/gradient_test.dart
Comment (nate-thegrate) [inline packages/flutter/test/painting/gradient_test.dart:451]: Most test descriptions don't include the word "test", since it's a bit redundant.

```suggestion
  test('LinearGradient withOpacity() preserves transform', () {
```

File: packages/flutter/test/painting/gradient_test.dart
Comment (nate-thegrate) [inline packages/flutter/test/painting/gradient_test.dart:794]:
```suggestion
  test('RadialGradient withOpacity() preserves transform', () {
```

File: packages/flutter/test/painting/gradient_test.dart
Comment (nate-thegrate) [inline packages/flutter/test/painting/gradient_test.dart:1056]:
```suggestion
  test('SweepGradient withOpacity() preserves transform', () {
```

**Review by nate-thegrate — APPROVED** [review]
I wanted to make a tiny wording suggestion, but otherwise this LGTM 👍

Thanks for catching & fixing this bug!

**Comment by hyhugh** [comment]
feel free to merge the PR, just realised i don't have access to submit the cl :)

**Comment by nate-thegrate** [comment]
> feel free to merge the PR, just realised i don't have access to submit the cl :)

Sounds good! Per the [Tree Hygiene](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#getting-a-code-review) page, we'll be able to merge once we have 2 approvals. I'll send out a message today to see if we can get a second stamp here :)

**Review by Piinks — APPROVED** [review]
image

**Pull request merged**
